### PR TITLE
Enabling soft commits for Solr4

### PR DESCRIPTION
**Pull request by hammady**
This pull request adds the ability to do soft commits in Solr4. Three function pairs were patched: commit, commit_if_dirty and commit_if_delete_dirty in Sunspot and Session classes. An optional boolean parameter denotes a soft commit if true. New tests are added and all tests are passing. Documentation is also updated to explain hard and soft commits.

### CODE REVIEW

**Comment by jonathanhoskin** [comment]
Is there a plan to merge this? We too are wanting to use soft commits in our app.

**Comment by lucygeneric** [comment]
+1

**Comment by joneslee85** [comment]
:+1:

File: sunspot/lib/sunspot.rb [hunk 1]
```diff
@@ -510,10 +514,10 @@ def dirty?
     end
 
     # 
-    # Sends a commit if the session is dirty (see #dirty?).
+    # Sends a commit (soft or hard) if the session is dirty (see #dirty?).
     #
-    def commit_if_dirty
-      session.commit_if_dirty
+    def commit_if_dirty(softCommit = false)
```
Comment (joneslee85) [inline sunspot/lib/sunspot.rb]: can we use snake case naming here instead? that is `soft_commit=false`?

**Comment by jonathanhoskin** [comment]
Is this PR going to get merged soon?

**Comment by njakobsen** [comment]
If the references to ```softCommit``` are changed to snake case we can get this merged

**Comment by hammady** [comment]
I have applied the snake case, ready to merge.

**Comment by hammady** [comment]
Yay thanks :-)

**Pull request merged**